### ISSUE TITLE
Requesting for the pretrained model (.pth) 

### ISSUE THREAD

**Issue by phanindrasp** [Mohamedelrefaie/DrivAerNet#16]
Hello,

I have been trying to replicate the results of DGCNN for the drivaernet dataset. However, as the earlier issues have mentioned, even though the loss is  reducing, R2 is consistently negative/close to zero (never goes above zero). I took the exact same hyperparameters as in the drivaernet paper except for batch size, which was chosen as 8. In this regard, I request you to share a pretrained model (.pth file) which would help me validate and possibly debug my model.

Thanks,
Phani.

**Comment by JJ-interesting** [comment]
Have you resolved the issue with the R2 score? If so, could you share the solution?

**Comment by phanindrasp** [comment]
Hey, actually not yet..Still waiting for the authors to response :(

**Comment by JJ-interesting** [comment]
Did you manage to reproduce the results from the paper? If so, could you share them? These are the results I obtained from my training. The R² score has turned positive but hasn't reached the 0.901 reported in the paper.
Total training time: 11301.56s
Model saved to models/cdPrediction DrivAerNet r2 100epochs 5k final model.pth
Testing the final model:
Test MSE: 0.000292, TeSt MAE: 0.013641, MaX MAE: 0.022632, Test R2: 0.5032
Total inference time: 1.21s for 592 samples
Testing the best model:
Test MSE: 0.000229, TeSt MAE: 0.011788, MaX MAE: 0.022385, Test R2: 0,6315
Total inference time: 1,10s for 592 samples

**Comment by phanindrasp** [comment]
I couldn't manage to bring R^2 above zero at all! What changes did you make that made your R^2 positive? Could you share a few other details from your simulation:
1. Batch size
2. Number of epochs
Could you also share your email ID so we can discuss this further? Mine is phanindraspk@gmail.com. Please feel free to mail.

Thanks,
Phanindra.

**Comment by JJ-interesting** [comment]
Okay, here’s my email: interesting55280@gmail.com. I’ll send the detailed information to your email later.

**Comment by LilaKen** [comment]
> Did you manage to reproduce the results from the paper? If so, could you share them? These are the results I obtained from my training. The R² score has turned positive but hasn't reached the 0.901 reported in the paper. Total training time: 11301.56s Model saved to models/cdPrediction DrivAerNet r2 100epochs 5k final model.pth Testing the final model: Test MSE: 0.000292, TeSt MAE: 0.013641, MaX MAE: 0.022632, Test R2: 0.5032 Total inference time: 1.21s for 592 samples Testing the best model: Test MSE: 0.000229, TeSt MAE: 0.011788, MaX MAE: 0.022385, Test R2: 0,6315 Total inference time: 1,10s for 592 samples

As you mentioned, the R² score we calculated is quite similar to your results. We also found that it's difficult to fully reproduce the performance reported in the original paper for the DrivAerNet dataset. We believe this is mainly due to significant differences between the predicted and true values for some samples, and common metrics such as R² do not capture these discrepancies well.

We later validated our hypothesis by comparing the predicted values to the true values, and we found the relative error to be 0.0748, which is much better than the 13% reported in the original paper. In summary, the differences in the results are mainly due to the characteristics of the data and the specific metrics used.

If you have any further questions, I would be happy to discuss and help out!
